ui/raidboss/data/06-ew/dungeon/the_lunar_subterrane.ts

```diff
@@ -162,6 +172,111 @@ const triggerSet: TriggerSet<Data> = {
       netRegex: { id: '88C2', source: 'Durante' },
       response: Responses.tankBuster(),
     },
+    {
+      // Round 1 is always a non-splitting line of orbs.
+      id: 'Lunar Subterrane Durante Forsaken Fount 1',
+      type: 'Ability',
+      netRegex: { id: '88BB', source: 'Durante', capture: false },
+      condition: (data) => data.fountsSeen === 0,
+      delaySeconds: 1, // This collides with Fount 2 if we don't delay. ???
+      infoText: (_data, _matches, output) => output.avoid!(),
+      run: (data) => data.fountsSeen += 1,
+      outputStrings: {
+        avoid: {
+          en: 'Away from orbs',
+        },
+      },
+    },
+    {
+      // Round 2 is always a splitting line. Center is always safe.
+      id: 'Lunar Subterrane Durante Forsaken Fount 2',
+      type: 'Ability',
+      netRegex: { id: '88BB', source: 'Durante', capture: false },
+      condition: (data) => data.fountsSeen === 1,
+      delaySeconds: 1,
+      response: Responses.goMiddle('info'),
+      run: (data) => data.fountsSeen += 1,
+    },
+    {
+      // On round three and subsequently, known  spawn locations for the Aetheric Charge orbs are:
+      // (0,-422)
+      // (0, -434.8)
+      // (0, -409.2)
+      // (3.65, -412.9)
+      // (-3.65, -412.95)
+      // (9.05, -412.8)
+      // (-9.05, -418.35)
+      // (9.15, -425.55)
+      // (9.15, -431.1)
+      // (-9.25, -431.1)
+      // (12.8, -422)
+      // (-12.8, -422)
+      // (Other spawn locations exist for rounds 1/2, but we can ignore those.)
+      id: 'Lunar Subterrane Durante Forsaken Fount 3 Collect',
+      type: 'AddedCombatant',
+      netRegex: { name: 'Aetheric Charge' },
+      condition: (data) => data.fountsSeen === 2,
+      run: (data, matches) => {
+        // Because the positions are relatively fixed, we don't need a reliable order for coordinates.
+        // Only the values and the count of those values really is important here.
+        data.fountX.push(Math.round(parseFloat(matches.x)));
+        data.fountY.push(Math.round(parseFloat(matches.y)) + 422); // Normalize the Y axis to 0
+      },
+    },
+    {
+      id: 'Lunar Subterrane Durante Forsaken Fount 3 Call',
+      type: 'AddedCombatant',
+      netRegex: { name: 'Aetheric Charge', capture: false },
+      condition: (data) => data.fountsSeen === 2,
+      delaySeconds: 0.5,
+      infoText: (data, _matches, output) => {
+        if (data.fountX.length < 5 || data.fountY.length < 5)
+          return;
+        // If there are five orbs on the field, three of them will, guaranteed,
+        // have the same X or Y value. Those three are in a straight line.
+        const hCount = data.fountY.filter((n) => n === 0).length;
+        const vCount = data.fountX.filter((n) => n === 0).length;
```

Comment:
Can you add some tolerance here for "close to zero"? This is maybe me being conservative but I think it just feels wrong to do `===` on a floating point value. Even `Math.abs(n) < 1` seems plenty.

Reply:
It's not a float, because `Math.round` returns integers, but I suppose we can still do that to avoid potential issues with -0 if the `matches` value ends up being in the -0.5 - 0.0 range.

Reply:
Oh quite right, this is all rounded. sounds good.